homeassistant/components/rainmachine/sensor.py

```diff
@@ -306,7 +352,21 @@ def update_from_latest_data(self) -> None:
             )
 
             if clicks and clicks_per_m3:
-                self._attr_native_value = (clicks * 1000) / clicks_per_m3
+                self._attr_native_value = round((clicks * 1000) / clicks_per_m3, 1)
+            else:
+                self._attr_native_value = None
+        elif self.entity_description.key == TYPE_FLOW_SENSOR_LEAK_CLICKS:
+            self._attr_native_value = self.coordinator.data.get("system", {}).get(
+                "flowSensorLeakClicks"
+            )
+        elif self.entity_description.key == TYPE_FLOW_SENSOR_LEAK_VOLUME:
+            clicks = self.coordinator.data.get("system", {}).get("flowSensorLeakClicks")
+            clicks_per_m3 = self.coordinator.data.get("system", {}).get(
+                "flowSensorClicksPerCubicMeter"
+            )
+
+            if clicks and clicks_per_m3:
+                self._attr_native_value = round((clicks * 1000) / clicks_per_m3, 1)
```

Comment:
Feels like we have a bunch of replicated logic throughout here – what would you think about some easy helper methods (attached to this class) that centralize it?

I would look at anything you're replicating between sensor types (e.g., the `round((clicks * 1000) / clicks_per_m3, 1)` calculation [and the data points fed into it]).

Reply:
Restructured the method to be less redundant. Take a look.

Reply:
Looks reasonable to me!